src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -58,8 +58,18 @@ public void onBlockPlaceExisting(BlockPlaceEvent e) {
         // This prevents Players from placing a block where another block already exists
         // While this can cause ghost blocks it also prevents them from replacing grass
         // or saplings etc...
-        if (BlockStorage.hasBlockInfo(e.getBlock())) {
-            e.setCancelled(true);
+        Block block = e.getBlock();
+        SlimefunItem sfItem = BlockStorage.check(block);
+
+        if (sfItem != null) {
+            if (e.getBlockReplacedState().getType().isAir()) {
```

Comment:
As stated in a prior review, these two checks should be swapped for performance reason.
Any `BlockStorage` operations might be heavy depending on their concurrent behaviour and take a relatively long time to compute, whereas the `.isAir()` check is very inexpensive.
Swapping will reduce those heavy lookups to a bare minimum.

```suggestion
        if (e.getBlockReplacedState().getType().isAir()) {
            Block block = e.getBlock();
            SlimefunItem sfItem = BlockStorage.check(block);

            if (sfItem != null) {
```